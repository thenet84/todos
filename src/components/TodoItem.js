import React from 'react';
import PropTypes from 'prop-types';

export const TodoItem = ({ id, completed, deleteTodo, markComplete, title }) => (
  <div className="todoItemWrapper">
    <p className={completed ? 'todoItemCompleted' : 'todoItem'}>
      <input type="checkbox" defaultChecked={completed} onChange={markComplete(id)} /> {title}
      <button type="button" className="deleteButton" onClick={deleteTodo(id)}>
        X
      </button>
    </p>
  </div>
);

TodoItem.propTypes = {
  id: PropTypes.number,
  completed: PropTypes.bool,
  deleteTodo: PropTypes.func,
  title: PropTypes.string,
  markComplete: PropTypes.func,
};
