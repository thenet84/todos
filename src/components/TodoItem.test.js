import React from 'react';
import { shallow } from 'enzyme';

import { TodoItem } from './TodoItem';

describe('TodoItem Component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<TodoItem />);
    expect(wrapper).toMatchSnapshot();
  });
});
