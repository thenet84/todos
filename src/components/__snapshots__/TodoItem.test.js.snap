// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem Component should render correctly 1`] = `<h3 />`;
